src/dpu/dpu-jpeg-decode.c

```diff
@@ -5,7 +5,7 @@
 #include "jpeg-host.h"
 #include "dpu-jpeg.h"
 
-__mram_noinit short MCU_buffer[NR_TASKLETS][MEGABYTE(16) / NR_TASKLETS];
+__mram_noinit short MCU_buffer[NR_TASKLETS][MEGABYTE(23) / NR_TASKLETS];
```

Comment:
Where did you get 23 megabytes from? That is considerably shorter than 16M shorts (=32 MB)